docs/elasticsearch-spec.asciidoc

```diff
@@ -3,24 +3,65 @@
 
 There are a number of settings which need to be considered before going into production related to Elasticsearch but also to Kubernetes.
 
-Basic settings
+**<<{p}-basic-settings>>**
 
-- JVM heap size
-- Node configuration
-- HTTP settings & TLS SANs
-- Resource limits
-- Pod Template
-- Volume claim templates
+- <<{p}-pod-template>>
+- <<{p}-node-configuration>>
+- <<{p}-volume-claim-templates>>
+- <<{p}-http-settings-tls-sans>>
 
-Advanced settings
+**<<{p}-advanced-settings>>**
+
+- <<{p}-virtual-memory>>
+- <<{p}-custom-http-certificate>>
+- <<{p}-es-secure-settings>>
+- <<{p}-bundles-plugins>>
+- <<{p}-init-containers-plugin-downloads>>
+- <<{p}-update-strategy>>
+  - <<{p}-change-budget>>
+- <<{p}-group-definitions>>
+
+[id="{p}-basic-settings"]
+== Basic settings
+
+[id="{p}-pod-template"]
+=== Pod Template
+
+Pod templates are the same pod templates you know and love from stateful sets and deployments. You can provide your own to add new settings, or merge settings with our defaults. For instance, if you want to add new labels to your pods, you can apply a pod template like so:
+
+[source,yaml]
+----
+    podTemplate:
+      metadata:
+        labels:
+          # additional labels for pods
+          foo: bar
+----
+
+More common is setting resource requests and limits, which is covered in link:managing-compute-resources.asciidoc[Managing compute resources].
+
+You may want to also install additional plugins, which is described in the <<{p}-init-containers-plugin-downloads>> section.
+
+You may want to set environment variables to configure Elasticsearch. For instance to set the minimum and maximum JVM heap size to `2g` and `4g` respectively you may modify the environment variables of the `elasticsearch` container as follows:
+
```

Comment:
Maybe we should merge #1348 first and refer to it here to show how to set env vars.

Reply:
As Seb said, let's merge and fix overlaps in follow-up PRs.